### ISSUE TITLE
Hiearchy Change URL doesnt work for subdirectory installation

### ISSUE THREAD

**Issue by almostengr** [farmOS/farmOS#56]
If FarmOS is installed within a subdirectory, instead of being on a dedicated server, the Hierarchy Change URL navigates to http://domain/admin/structure/taxonomy/farm_areas when it should navigate to http://domain/subdirectory/admin/structure/taxonomy/farm_areas. 

In the lower left of the attached screenshot, it shows that the change link navigates to "http://thepost/?q=admin/structure/taxonomy/farm_areas&destination=farm/areas" when it should navigate to (based on the URL in the browser) "http://thepost/farmosweb?q=admin/structure/taxonomy/farm_areas&destination=farm/areas". 

Not sure which line is causing this problem. I've narrowed it down to lines 49 and 155. I believe that line is 155 that's the problem. Removing the leading slash (before admin) in the URL should fix this.

https://github.com/farmOS/farmOS/blob/4dc82b53dd5f01fed122ff5e3267f50f37d473d5/modules/farm/farm_area/farm_area.views_default.inc#L49

https://github.com/farmOS/farmOS/blob/4dc82b53dd5f01fed122ff5e3267f50f37d473d5/modules/farm/farm_area/farm_area.views_default.inc#L155

![screenshot from 2018-03-01 08-20-52](https://user-images.githubusercontent.com/15150637/36849364-7fa16e0e-1d29-11e8-8f54-ec011a5fcb96.png)

**Comment by mstenta** [comment]
Ah good catch @bitsecondal ! That should be easy to fix... I'll take a look right now.

**Comment by mstenta** [comment]
Ok interesting...

Line 49 appears to be correct:

`  $handler->display->display_options['header']['area_text_custom']['content'] = '<h4>Hierarchy <small>' . l('(change)', 'admin/structure/taxonomy/farm_areas', array('query' => array('destination' => 'farm/areas'))) . '</small></h4>';`

But line 155 is not:

`    t('<h4>Hierarchy <small><a href="/admin/structure/taxonomy/farm_areas?destination=farm/areas">(change)</a></small></h4>'),`

To explain what both of those are: line 49 is defining header text for the View, and it's using the `l()` function to generate a link to the page for changing the hierarchy of the taxonomy.

The `l()` function will automatically handle cases where Drupal is installed in a subdirectory, as far as I know. So it SHOULD be working.

Line 155 is simply the translated version of that string (which only gets added to the code if the Locale module is installed).  As far as I know, it doesn't get used at all, unless you are using translations.  But I could be wrong about that.

Can you try something for me?

Try changing line 155 to the following:

`    t('<h4>Hierarchy <small>' . l('(change)', 'admin/structure/taxonomy/farm_areas', array('query' => array('destination' => 'farm/areas'))) . '</small></h4>'),`

Then reload your areas page and see if that fixes it.  I kinda doubt that it will... but if it does then that's great!  Ultimately, those two strings SHOULD match... so that's a small bug regardless.  But I'm not sure if that's what's causing THIS bug specifically... let's see!

**Comment by almostengr** [comment]
I'm not using translations but it does appear that it is using that string (not sure why).

I updated line 155 as mentioned and it now works.  I've attached the screenshots.

![screenshot from 2018-03-01 08-53-38](https://user-images.githubusercontent.com/15150637/36851102-1266c424-1d2e-11e8-9cad-b7284ca02063.png)
![screenshot from 2018-03-01 08-56-05](https://user-images.githubusercontent.com/15150637/36851240-6a41f664-1d2e-11e8-973f-80e62ae0de55.png)
